### ISSUE TITLE
recording metric can lead to java.lang.ArithmeticException: integer overflow

### ISSUE THREAD

**Issue by chriz-active** [active-group/active-logger#7]
```
(ns active.clojure.logger.benchmark.metric-accumulator
  (:require [active.clojure.logger.metric-accumulator      :as m]))

(defn working-record-ang-get!
  []
  (let [histogram-metric (m/make-histogram-metric "name" "help" [5])
        metric-store     (m/fresh-metric-store)]

    (println (m/record-and-get! metric-store histogram-metric {} 999999999999999999999999 0))
    (println (m/record-and-get! metric-store histogram-metric {} 999999999999999999999999 0))))

(defn failing-record-and-get!
  []
  (let [histogram-metric (m/make-histogram-metric "name" "help" [5])
        metric-store     (m/fresh-metric-store)]

    (println (m/record-and-get! metric-store histogram-metric {} 8999999999999999991 0))
    (println (m/record-and-get! metric-store histogram-metric {}  999999999999999999 0))))

(println "WORKING-------------------------------------")
(working-record-ang-get!)
(println "FAILING-------------------------------------")
(failing-record-and-get!)
```
Output

```
WORKING-------------------------------------
(#active.clojure.logger.metric_accumulator.MetricSample{:name name_sum, :labels {}, :value 999999999999999999999999N, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_count, :labels {}, :value 1, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le +Inf}, :value 1, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le 5}, :value 0, :timestamp 0})
(#active.clojure.logger.metric_accumulator.MetricSample{:name name_sum, :labels {}, :value 1999999999999999999999998N, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_count, :labels {}, :value 2, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le +Inf}, :value 2, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le 5}, :value 0, :timestamp 0})
FAILING-------------------------------------
(#active.clojure.logger.metric_accumulator.MetricSample{:name name_sum, :labels {}, :value 8999999999999999991, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_count, :labels {}, :value 1, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le +Inf}, :value 1, :timestamp 0} #active.clojure.logger.metric_accumulator.MetricSample{:name name_bucket, :labels {:le 5}, :value 0, :timestamp 0})
Exception in thread "main" Syntax error compiling at (active/clojure/logger/benchmark/metric_accumulator.clj:23:1).
	at clojure.lang.Compiler.load(Compiler.java:7647)
	at clojure.lang.RT.loadResourceScript(RT.java:381)
	at clojure.lang.RT.loadResourceScript(RT.java:372)
	at clojure.lang.RT.load(RT.java:463)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5985)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at user$eval224.invokeStatic(form-init6806175156840149216.clj:1)
	at user$eval224.invoke(form-init6806175156840149216.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7176)
	at clojure.lang.Compiler.eval(Compiler.java:7166)
	at clojure.lang.Compiler.load(Compiler.java:7635)
	at clojure.lang.Compiler.loadFile(Compiler.java:7573)
	at clojure.main$load_script.invokeStatic(main.clj:452)
	at clojure.main$init_opt.invokeStatic(main.clj:454)
	at clojure.main$init_opt.invoke(main.clj:454)
	at clojure.main$initialize.invokeStatic(main.clj:485)
	at clojure.main$null_opt.invokeStatic(main.clj:519)
	at clojure.main$null_opt.invoke(main.clj:516)
	at clojure.main$main.invokeStatic(main.clj:598)
	at clojure.main$main.doInvoke(main.clj:561)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.main.main(main.java:37)
Caused by: java.lang.ArithmeticException: integer overflow
	at clojure.lang.Numbers.throwIntOverflow(Numbers.java:1576)
	at clojure.lang.Numbers.add(Numbers.java:1893)
	at clojure.lang.Numbers$LongOps.add(Numbers.java:485)
	at clojure.lang.Numbers.add(Numbers.java:153)
	at clojure.core$_PLUS_.invokeStatic(core.clj:992)
	at clojure.core$_PLUS_.invoke(core.clj:984)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at active.clojure.lens$overhaul.invokeStatic(lens.cljc:108)
	at active.clojure.lens$overhaul.doInvoke(lens.cljc:102)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at active.clojure.logger.metric_accumulator$update_metric_value.invokeStatic(metric_accumulator.clj:122)
	at active.clojure.logger.metric_accumulator$update_metric_value.invoke(metric_accumulator.clj:115)
	at active.clojure.logger.metric_accumulator$inc_metric_value$fn__2545.invoke(metric_accumulator.clj:136)
	at clojure.core$update.invokeStatic(core.clj:6196)
	at clojure.core$update.invoke(core.clj:6188)
	at active.clojure.logger.metric_accumulator$inc_metric_value.invokeStatic(metric_accumulator.clj:134)
	at active.clojure.logger.metric_accumulator$inc_metric_value.invoke(metric_accumulator.clj:131)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at active.clojure.lens$overhaul.invokeStatic(lens.cljc:108)
	at active.clojure.lens$overhaul.doInvoke(lens.cljc:102)
	at clojure.lang.RestFn.invoke(RestFn.java:494)
	at active.clojure.logger.metric_accumulator$update_histogram_values.invokeStatic(metric_accumulator.clj:403)
	at active.clojure.logger.metric_accumulator$update_histogram_values.invoke(metric_accumulator.clj:394)
	at active.clojure.logger.metric_accumulator$update_stored_values.invokeStatic(metric_accumulator.clj:490)
	at active.clojure.logger.metric_accumulator$update_stored_values.invoke(metric_accumulator.clj:480)
	at active.clojure.logger.metric_accumulator$record_metric_1$fn__2906.invoke(metric_accumulator.clj:588)
	at clojure.core$update.invokeStatic(core.clj:6196)
	at clojure.core$update.invoke(core.clj:6188)
	at active.clojure.logger.metric_accumulator$record_metric_1.invokeStatic(metric_accumulator.clj:586)
	at active.clojure.logger.metric_accumulator$record_metric_1.invoke(metric_accumulator.clj:584)
	at clojure.lang.AFn.applyToHelper(AFn.java:165)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Atom.swap(Atom.java:79)
	at clojure.core$swap_BANG_.invokeStatic(core.clj:2355)
	at clojure.core$swap_BANG_.doInvoke(core.clj:2345)
	at clojure.lang.RestFn.invoke(RestFn.java:497)
	at active.clojure.logger.metric_accumulator$record_metric_BANG_.invokeStatic(metric_accumulator.clj:607)
	at active.clojure.logger.metric_accumulator$record_metric_BANG_.invoke(metric_accumulator.clj:598)
	at active.clojure.logger.metric_accumulator$record_and_get_BANG_.invokeStatic(metric_accumulator.clj:913)
	at active.clojure.logger.metric_accumulator$record_and_get_BANG_.invoke(metric_accumulator.clj:907)
	at active.clojure.logger.benchmark.metric_accumulator$failing_record_and_get_BANG_.invokeStatic(metric_accumulator.clj:18)
	at active.clojure.logger.benchmark.metric_accumulator$failing_record_and_get_BANG_.invoke(metric_accumulator.clj:12)
	at active.clojure.logger.benchmark.metric_accumulator$eval3161.invokeStatic(metric_accumulator.clj:23)
	at active.clojure.logger.benchmark.metric_accumulator$eval3161.invoke(metric_accumulator.clj:23)
	at clojure.lang.Compiler.eval(Compiler.java:7176)
	at clojure.lang.Compiler.load(Compiler.java:7635)
	... 41 more
```

https://github.com/active-group/active-logger/blob/0ed314983471bff24340c11baea5d8d747cb06a4/src/active/clojure/logger/metric_accumulator.clj#L136

Might be fixed using `+'` instead of `+`

**Comment by cresh** [comment]
Please also add tests for those Bignums in `active.clojure.logger.metric-prometheus-test`.

**Comment by chriz-active** [comment]
alternative discussed offline: use `double` and change the spec to expect doubles

**Comment by chriz-active** [comment]
fixed by 3bfc56c858466cc612f89afdf4e7da60bb0c3c8a
